src/ensembl/src/content/app/entity-viewer/shared/rest/rest-data-fetchers/transcriptData.ts

```diff
@@ -77,18 +79,21 @@ export const fetchTranscript = async (
   signal?: AbortSignal
 ): Promise<Transcript> => {
   const transcriptUrl = `https://rest.ensembl.org/lookup/id/${id}?expand=1;content-type=application/json`;
-  const transcript: TranscriptInResponse = (await fetch(transcriptUrl, {
-    signal
-  }).then((response) => response.json())) as TranscriptInResponse;
+  const transcript: TranscriptInResponse = (await apiService.fetch(
+    transcriptUrl,
+    {
+      signal
+    }
+  )) as TranscriptInResponse;
 
   let proteinFeatures;
 
-  if (transcript.Translation) {
+  if (transcript?.Translation) {
```

Comment:
don't you promise on line 87 that transcript is of type `TranscriptInResponse`? Why the question mark then?

Reply:
That's right. I temporarily added it as the rest endpoint kept breaking for me (most likely to be connection issue on my side). I've removed it.